OpenRA.Test/OpenRA.Mods.Common/ShapeTest.cs

```diff
@@ -0,0 +1,95 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using NUnit.Framework;
+using OpenRA.Mods.Common.HitShapes;
+
+namespace OpenRA.Test
+{
+	[TestFixture]
+	public class ShapeTest
+	{
+		IHitShape shape;
+
+		[TestCase(TestName = "CircleShape reports accurate distance")]
+		public void Circle()
+		{
+			shape = new CircleShape(new WDist(1234));
+			shape.Initialize();
+
+			Assert.That(shape.DistanceFromEdge(new WVec(100, 100, 0)).Length,
+				Is.EqualTo(0));
+
+			Assert.That(shape.DistanceFromEdge(new WVec(1000, 0, 0)).Length,
+				Is.EqualTo(0));
+
+			Assert.That(shape.DistanceFromEdge(new WVec(2000, 2000, 0)).Length,
+				Is.InRange(1593, 1595));
+
+			Assert.That(new CircleShape(new WDist(73))
+				.DistanceFromEdge(new WVec(150, -100, 0)).Length,
+				Is.InRange(106, 108));
+
+			Assert.That(new CircleShape(new WDist(55))
+				.DistanceFromEdge(new WVec(30, -45, 0)).Length,
+				Is.InRange(-1, 1));
```

Comment:
I swear there was an earlier comment about this, but I agree that these tests should specify exact values and not ranges. The output must be deterministic or else we'll get desyncs.